File: cpp/Estruturas repetitivas/media_idades.cpp
#include <iostream>
using namespace std;

int main() {

    int idade, quantidade = 0;
    double soma = 0, media;

    cout << "Digite as idades:" << endl;
    cin >> idade;

    while (idade >= 0) {

        soma += idade;
        quantidade++;

        cin >> idade;
    }

    if (quantidade == 0) {
        cout << "IMPOSSIVEL CALCULAR" << endl;
    }
    else {
        media = soma / quantidade;
        cout << "MEDIA = " << media << endl;
    }

    return 0;
}

File: java/Estruturas repetitivas/MediaIdades.java
import java.util.Locale;
import java.util.Scanner;

public class MediaIdades {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int idade;
        int soma = 0;
        int quantidade = 0;
        double media;

        System.out.print("Digite uma idade: ");
        idade = sc.nextInt();

        while (idade >= 0) {
            soma += idade;
            quantidade++;

            System.out.print("Digite uma idade: ");
            idade = sc.nextInt();
        }

        if (quantidade == 0) {
            System.out.println("IMPOSSIVEL CALCULAR");
        } else {
            media = (double) soma / quantidade;
            System.out.printf("MEDIA = %.2f%n", media);
        }

        sc.close();
    }
}